### DISCUSSION TITLE
print('Value={:3.30f}'.format(12.34567))問題

### DISCUSSION THREAD

**Discussion by EricAgumonJou** [1101-GS4719-C/Python-Community-GS4719-C#87] [General]
剛剛輸入:
print('Value={:3.30f}'.format(12.34567))
輸出為:
Value=12.345670000000000143813849717844

我想問一下輸出的143813849717844是亂碼嗎?

**Comment by enoch-lightning** [comment]
就我目前的理解，這是必然的結果，稱為「浮點誤差」，也就是在計算浮點數時必定存在的誤差。你由{:3.30f}將小數點後增加至30位時，就看出來了。

**Reply by EricAgumonJou** [reply]
謝謝你~發現新世界 : )

**Comment by jerry850313** [comment]
我覺得你可以參考[這篇](https://docs.python.org/zh-tw/3/tutorial/floatingpoint.html)

**Reply by jerry850313** [reply]
然後配[這個](https://edisonx.pixnet.net/blog/post/46103946)服用，簡單來說，電腦將12.34567轉成2進位存取，然後自然會有後面那串，但是電腦幫你把後面那串弄成不影響計算的樣子。(以下是我幫你取到小數點後100位的結果)
![螢幕擷取畫面 2021-10-18 010954](https://user-images.githubusercontent.com/56184165/137637669-1be6b904-f999-4240-9211-c3b18bf95a32.jpg)

**Reply by jerry850313** [reply]
想進一步了解二進位可以進來
[傳送門](https://kknews.cc/zh-tw/invest/qlm8m4g.html)

**Reply by EricAgumonJou** [reply]
感謝你~真的好神奇,沒想到二進位會這樣處理

**Reply by jerry850313** [reply]
電腦會採二進位是因為，區分高電位(High)和低電位(Low)比起把電壓分成十等分再判斷在哪個區間容易的許多，而且採取二進位運算也比較不容易判斷錯，因此一般電腦和計算機都是採用二進位運算。但以上所述，數位系統導論和計算機概論都有講過吧?

**Reply by EricAgumonJou** [reply]
有，我想起來了，跟IEEE二進位浮點數算術標準（IEEE 754）有關係

**Reply by EricAgumonJou** [reply]
補充給有興趣的同學：
https://www.geeksforgeeks.org/ieee-standard-754-floating-point-numbers/
